camayoc/tests/qpc/cli/test_sources.py

```diff
@@ -286,6 +284,304 @@ def test_add_with_port_negative(
+    cred_add_and_check(
+        {
+            'name': cred_name,
+            'username': utils.uuid4(),
+            'password': None,
+            'type': source_type,
+        },
+        [(CONNECTION_PASSWORD_INPUT, utils.uuid4())],
+    )
+
+    qpc_source_add = pexpect.spawn(
+        'qpc source add --name {} --cred {} --hosts {} --ssl-protocol {} '
+        '--type {}'
+        .format(name, cred_name, hosts, ssl_protocol, source_type)
+    )
+    assert qpc_source_add.expect('Source "{}" was added'.format(name)) == 0
+    assert qpc_source_add.expect(pexpect.EOF) == 0
+    qpc_source_add.close()
+    assert qpc_source_add.exitstatus == 0
+
+    source_show_and_check(
+        {'name': name},
+        generate_show_output({
+            'cred_name': cred_name,
+            'hosts': hosts,
+            'name': name,
+            'options': {
+                'ssl_protocol': ssl_protocol,
+            },
+            'port': default_port_for_source(source_type),
+            'source_type': source_type,
+        })
+    )
+
+
+def test_add_with_ssl_protocol_negative(
+        isolated_filesystem, qpc_server_config, source_type):
+    """Try to add source with cred, hosts and an invalid ssl_protocol.
+
+    :id: 2ccad703-a9b5-46d0-a717-03aa4df23246
+    :description: Add a source entry providing the ``--name``, ``--cred``,
+        ``--hosts`` and ``--ssl-protocol`` options. The value for the
+        ``--ssl-protocol`` should be an invalid value, for example a
+        different value from ``true`` or ``false`` for host manager types and
+        the option should not be valid for network type.
+    :steps: Run ``qpc source add --name <name> --cred <cred> --hosts <hosts>
+        --port <port> --type <type>``
+    :expectedresults: An error message is printed and a non-zero status code
+        should be returned. Also no source entry is created.
+    """
+    cred_name = utils.uuid4()
+    hosts = '127.0.0.1'
+    name = utils.uuid4()
+    port = default_port_for_source(source_type)
+    ssl_protocol = utils.uuid4()
+    cred_add_and_check(
+        {
+            'name': cred_name,
+            'username': utils.uuid4(),
+            'password': None,
+            'type': source_type,
+        },
+        [(CONNECTION_PASSWORD_INPUT, utils.uuid4())],
+    )
+
+    qpc_source_add = pexpect.spawn(
+        'qpc source add --name {} --cred {} --hosts {} --port {} '
+        '--ssl-protocol {} --type {}'
+        .format(name, cred_name, hosts, port, ssl_protocol, source_type)
+    )
+    assert qpc_source_add.expect(
+        'qpc source add: error: argument --ssl-protocol: invalid choice: '
+        '\'{}\' \\(choose from \'SSLv23\', \'TLSv1\', \'TLSv1_1\', '
+        '\'TLSv1_2\'\\)'
+        .format(ssl_protocol)
+    ) == 0
+    assert qpc_source_add.expect(pexpect.EOF) == 0
+    qpc_source_add.close()
+    assert qpc_source_add.exitstatus == 2
+
+
+# @pytest.mark.parametrize('source_type', QPC_HOST_MANAGER_TYPES)
```

Comment:
I will update the test once quipucords is fixed to not accept SSL options for network sources.